python/cudf/cudf/core/dataframe.py

```diff
@@ -1658,8 +1658,9 @@ def update(
         if not self.index.equals(other.index):
             other = other.reindex(self.index, axis=0)
 
-        for col in self.columns:
-            this = self[col]
+        source_df = self.copy(deep=False)
+        for col in source_df.columns:
```

Comment:
Instead of looping over `.column` could we use the `._column_names` property which avoids materializing a Pandas Index?